drivers/nfs.py

```diff
@@ -186,43 +182,75 @@ def unmount(mountpoint, rmmountpoint):
             raise NfsException("rmdir failed with error '%s'" % inst.strerror)
 
 
-def scan_exports(target):
+def scan_exports(target, transport):
     """Scan target and return an XML DOM with target, path and accesslist."""
     util.SMlog("scanning")
-    cmd = [SHOWMOUNT_BIN, "--no-headers", "-e", target]
     dom = xml.dom.minidom.Document()
     element = dom.createElement("nfs-exports")
     dom.appendChild(element)
-    for val in util.pread2(cmd).split('\n'):
-        if not len(val):
-            continue
-        entry = dom.createElement('Export')
-        element.appendChild(entry)
-
-        subentry = dom.createElement("Target")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(target)
-        subentry.appendChild(textnode)
-
-        # Access is not always provided by showmount return
-        # If none is provided we need to assume "*"
-        array = val.split()
-        path = array[0]
-        access = array[1] if len(array) >= 2 else "*"
-        subentry = dom.createElement("Path")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(path)
-        subentry.appendChild(textnode)
-
-        subentry = dom.createElement("Accesslist")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(access)
-        subentry.appendChild(textnode)
-
-    return dom
-
-
-def scan_srlist(path, dconf):
+    try:
```

Comment:
I don't particularly like how long this method has now become, it makes it quite cognitively hard to follow. Would be good to pull some of these long open coded blocks out into well named methods which describe what they're doing and then each one can be reviewed and understood in isolation (or ignored if not of interest to the current needs).

Reply:
Similarly, to a lesser extent, `get_supported_nfs_versions` below.

Reply:
I can split in methods for NFS3 and NFS4Only.

Reply:
Done :)